tianshou/data/batch.py

```diff
@@ -433,11 +442,11 @@ def __init__(
             # Feels like kwargs could be just merged into batch_dict in the beginning
             self.__init__(kwargs, copy=copy)  # type: ignore
 
-    def to_dict(self) -> dict[str, Any]:
+    def to_dict(self, recurse: bool = False) -> dict[str, Any]:
         result = {}
         for k, v in self.__dict__.items():
-            if isinstance(v, Batch):
-                v = v.to_dict()
+            if recurse and isinstance(v, Batch):
```

Comment:
This is also a breaking change since you changed the default to False. Not a problem, but pls add it to the changelog. I'm actually not sure False is the better default, what was your reason?

Reply:
The default of recursively convert to dict was correct before. I wanted to revert back my initial change of logic there, and I just forgot. Will add it now.